### DISCUSSION TITLE
What's the simplest way to do a two-file verify?

### DISCUSSION THREAD

**Discussion by paulsuart** [VerifyTests/Verify#1174] [Q&A]
I have a test that's currently verifying a record that has two properties and it's working fine.

This is the record I'm using:
```
public record CallResult(Object? Result, IEnumerable<string>? LogEntries);
```

The object in `Result` can be quite large, and the number of `LogEntries` can also be reasonably big leading to a large single verified file.

What I'd like is for the two properties to be saved as two separate `.verified.txt` files so the two unrelated bits of information can be viewed separately.  I have worked out how to create a custom converter, but I can't work out how to use the same serialisation settings as the standard call to `Verify` uses.

This is the converter:
```
        VerifierSettings.RegisterFileConverter<CallResult>((target, context) =>
        {
            var targets = new List<Target>();

            if (target.Result != null)
                targets.Add(new Target("txt", JsonSerializer.Serialize(<Need the current settings>, target.Result!), "Result"));

            if (target.LogEntries != null)
                targets.Add(new Target("txt",  JsonSerializer.Serialize(<Need the current settings>, target.LogEntries!), "SqlLogs"));

            return new ConversionResult(null, targets);
        });     
```

I would prefer not to make two separate calls to `Verify` if I can help it.

**Comment by SimonCropp** [comment]
hmm there is not really a good way to do that at the moment. ConversionResult assumes you have one object for serialization (the info param u r passing in null for) and a collection of targets (that you know how to serialize). 

i could possibly add a new constructor to target `new Target(object)` that will serialize the target. thoughts?

**Comment by paulsuart** [comment]
Thanks for adding your ideas, @SimonCropp.

This morning I settled on simply using `await Task.WhenAll` for multiple calls to `Verify` and that seems to work fine.
